ouroboros-network/src/Ouroboros/Network/Diffusion.hs

```diff
@@ -6,122 +6,152 @@
+           IOException   extraState extraState extraFlags extraPeers m
+    -> ExtraTracers 'P2P extraState extraFlags extraPeers m
+
+  P2PCardanoTracers
+    :: Common.TracersExtra
            RemoteAddress NodeToNodeVersion         NodeToNodeVersionData
            LocalAddress  NodeToClientVersion       NodeToClientVersionData
            IOException   CardanoPeerSelectionState CardanoPeerSelectionState
-           PeerTrustable (CardanoPublicRootPeers RemoteAddress) IO
-    -> ExtraTracers 'P2P
+           PeerTrustable (CardanoPublicRootPeers RemoteAddress) m
+    -> ExtraTracers 'P2PCardano CardanoPeerSelectionState PeerTrustable (CardanoPublicRootPeers RemoteAddress) m
 
   NonP2PTracers
     :: NonP2P.TracersExtra
-    -> ExtraTracers 'NonP2P
+    -> ExtraTracers 'NonP2P extraState extraFlags extraPeers m
 
 
 -- | Diffusion arguments which depend on p2p mode.
 --
-data ExtraArguments (p2p :: P2P) m where
+data ArgumentsExtra (p2p :: P2P) extraArgs extraPeers m where
   P2PArguments
-    :: P2P.ArgumentsExtra (CardanoArgumentsExtra m) PeerTrustable m
-    -> ExtraArguments 'P2P m
+    :: Common.ArgumentsExtra extraArgs extraPeers m
+    -> ArgumentsExtra 'P2P extraArgs extraPeers m
+
+  P2PCardanoArguments
+    :: Common.ArgumentsExtra (CardanoArgumentsExtra m) PeerTrustable m
+    -> ArgumentsExtra 'P2PCardano (CardanoArgumentsExtra m) PeerTrustable m
 
   NonP2PArguments
     :: NonP2P.ArgumentsExtra
-    -> ExtraArguments 'NonP2P m
-
+    -> ArgumentsExtra 'NonP2P extraArgs extraPeers m
 
 -- | Application data which depend on p2p mode.
 --
-data ExtraApplications (p2p :: P2P) ntnAddr m a where
+data Applications (p2p :: P2P) extraAPI m a where
   P2PApplications
-    :: P2P.ApplicationsExtra ntnAddr m a
-    -> ExtraApplications 'P2P ntnAddr m a
+    :: Common.Applications
+         RemoteAddress  NodeToNodeVersion   NodeToNodeVersionData
+         LocalAddress   NodeToClientVersion NodeToClientVersionData
+         (CardanoLedgerPeersConsensusInterface m) m a
+    -> Applications 'P2P extraAPI m a
+
+  P2PCardanoApplications
+    :: Common.Applications
+         RemoteAddress  NodeToNodeVersion   NodeToNodeVersionData
+         LocalAddress   NodeToClientVersion NodeToClientVersionData
+         (CardanoLedgerPeersConsensusInterface m) m a
+    -> Applications 'P2PCardano (CardanoLedgerPeersConsensusInterface m) m a
 
   NonP2PApplications
+    :: Common.Applications
+         RemoteAddress  NodeToNodeVersion   NodeToNodeVersionData
+         LocalAddress   NodeToClientVersion NodeToClientVersionData
+         () m a
+    -> Applications 'NonP2P () m a
+
+-- | Application data which depend on p2p mode.
+--
+data ApplicationsExtra (p2p :: P2P) ntnAddr m a where
+  P2PApplicationsExtra
+    :: Common.ApplicationsExtra ntnAddr m a
+    -> ApplicationsExtra 'P2P ntnAddr m a
+
+  P2PCardanoApplicationsExtra
+    :: Common.ApplicationsExtra ntnAddr m a
+    -> ApplicationsExtra 'P2PCardano ntnAddr m a
```

Comment:
And here as well.

ref: a382ed87d2998ff90dd825ddbc6ac2ccbdac8da0

Reply:
Same as above this disappears in another commit